sdk/core/Azure.Core.Amqp/src/AmqpMessageBody.cs

```diff
@@ -121,5 +151,40 @@ public bool TryGetSequence(out IEnumerable<IList<object>>? sequence)
             sequence = null;
             return false;
         }
+
+        private static void ValidateAmqpPrimitive(object value, string paramName)
+        {
+            Argument.AssertNotNull(value, paramName);
+            switch (value)
+            {
+                case string:
+                case byte:
+                case sbyte:
+                case char:
+                case short:
+                case ushort:
+                case int:
+                case uint:
+                case long:
+                case ulong:
+                case float:
+                case double:
+                case decimal:
+                case bool:
+                case Guid:
+                case DateTime:
+                    return;
+                case IEnumerable enumerable:
+                    foreach (object val in enumerable)
+                    {
+                        ValidateAmqpPrimitive(val, paramName);
```

Comment:
Why do you hate perf?

![https://media2.giphy.com/media/jYmGmDK3rKdkk/giphy.gif](https://media2.giphy.com/media/jYmGmDK3rKdkk/giphy.gif)